docs/docs.md

```diff
@@ -190,14 +219,38 @@ The KeyRegistry contract may need to be upgraded in case a bug is discovered or
 4. A new Bundler contract is deployed, pointing to the correct contracts.
 5. The contract is set to untrusted state where anyone can register keys.
 
-# 4. Bundler
+## 1.4. Bundler
 
 The Bundler contract lets a caller register an fid, rent storage units and register a key in a single transaction to save gas. It is a simple wrapper around contract methods and contains little logic beyond tracking contract addresses, collecting parameters and invoking the appropriate functions.
 
-# 5. Fname Resolver
+# 2. L1 Contracts
+
+The Fname Resolver contract is deployed on L1 Mainnet (chainid: 1).
+
+# 2.1. Fname Resolver
 
 The Fname Resolver contract validates usernames issued under the \*.fcast.id domain on-chain by implementing [ERC-3668](https://eips.ethereum.org/EIPS/eip-3668) and [ENSIP-10](https://docs.ens.domains/ens-improvement-proposals/ensip-10-wildcard-resolution). The resolver contains the url of the server which issues the usernames and proofs. It maintains a list of valid signers for the server and also validates proofs returned by the server.
 
 ### Administration
 
 An `owner` can update the list of valid signers associated with the server.
+
+# 3. Off-chain Systems
+
+# 3.1. Fname Registry
+
+The [Fname registry](https://github.com/farcasterxyz/fname-registry) is an off-chain server that lets addresses with an fid acquire a unique username. Requests are authenticated by way of a signed message produced by the fid's custody address.
+
+### Assumptions
+
+- All event history from IdRegistry will be accessible via an OP Mainnet node.
+- OP Mainnet will not re-org or revert transactions once confirmed by the sequencer.
```

Comment:
@sanjayprabhu what transaction safety level do we require before accepting a txn as safe?